virttest/utils_kernel_module.py

```diff
@@ -0,0 +1,151 @@
+        restore undoes loading.
+        :param params: Parameters to load with, 'key1=param1 key2=param2 ...'
+        """
+
+        if params == self._loaded_config:
+            logging.debug("Not reloading module, same parameters requested for"
+                          " module %s: %s" % (self._module_name, params))
+            return
+
+        if not force and not self._was_loaded:
+            logging.debug("Module %s isn't loaded. Set force=True to force"
+                          " loading." % self._module_name)
+        else:
+            cmd = ""
+            if self._was_loaded:
+                cmd += 'rmmod %s; ' % self._module_name
+            cmd += 'modprobe %s %s' % (self._module_name, params)
+            logging.debug("Loading module: %s" % cmd)
+            status, output = process.getstatusoutput(cmd, shell=True,
+                                                     ignore_status=True)
+            if status:
+                raise exceptions.TestError("Couldn't load module %s: %s" % (
+                    self._module_name, output
+                ))
+            else:
+                self._loaded_config = params
+
+        if self._persistence_file:
+            self.save()
+
+    def restore(self):
+        """Restore previous module state"""
+
+        if self._loaded_config:
+            if not self._was_loaded:
+                cmd = 'rmmod %s' % self._module_name
+            else:
+                cmd = 'rmmod %s; modprobe %s %s' % (self._module_name,
+                                                    self._module_name,
+                                                    self._config_backup)
+            logging.debug("Restoring module state: %s" % cmd)
+            status, output = process.getstatusoutput(cmd, shell=True,
+                                                     ignore_status=True)
+            if status:
+                raise exceptions.TestError("Couldn't restore module %s: %s" % (
+                    self._module_name, self._config_backup))
+
+        if self._persistence_file and \
+                os.path.isfile(self._persistence_file):
+            os.remove(self._persistence_file)
+
+    def _backup_config(self):
+        """
+        Check if self.module_name is loaded and read config
+
+        """
+        config = KernelModuleHandler._load_config(self._module_name)
+        if config:
+            self._config_backup = " ".join("%s=%s" % _ for _ in config.items())
+            self._was_loaded = True
+        else:
+            self._was_loaded = False
+        logging.debug("Backed up %s module state (was_loaded, params)"
+                      "=(%s, %s)" % (self._module_name, self._was_loaded,
+                                     self._config_backup))
+
+    def _load(self):
+        try:
+            with open(self._persistence_file, 'rb') as pickled:
+                self.__dict__.update(pickle.load(pickled))
```

Comment:
Suggestion: How about to dump only the `_config_backup` and `_was_loaded` as we don't really care about the others? Would be more explicit...